packages/flutter/lib/src/rendering/sliver_list.dart

```diff
@@ -90,8 +90,28 @@ class RenderSliverList extends RenderSliverMultiBoxAdaptor {
     // it's possible for a child to get removed without notice.
     RenderBox leadingChildWithLayout, trailingChildWithLayout;
 
-    // Find the last child that is at or before the scrollOffset.
     RenderBox earliestUsefulChild = firstChild;
+
+    // A firstChild with null layout offset is likely a result of children
+    // reordering.
+    //
+    // We rely on firstChild to have accurate layout offset. In the case of null
+    // layout offset, we have to find the first child that has valid layout
+    // offset.
+    if (childScrollOffset(firstChild) == null) {
+      int leadingChildrenWithoutLayoutOffset = 0;
+      while (childScrollOffset(earliestUsefulChild) == null) {
+        earliestUsefulChild = childAfter(firstChild);
```

Comment:
There is a small problem here： If I add multiple elements to the front at the same time, it may cause an endless loop.

```suggestion
        earliestUsefulChild = childAfter(earliestUsefulChild);
```

Reply:
oops, nice catch!, will fix it. thanks